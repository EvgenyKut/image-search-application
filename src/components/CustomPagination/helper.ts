export const paginationFormatter = (
  page: number,
  totalPages: number,
  type: 'increase' | 'decrease',
  value: number = 1,
) => {
  const positiveValue = page + value > totalPages ? totalPages : page + value;
  const negativeValue = page - value < 0 ? 0 : page - value;
  return type === 'increase' ? positiveValue : negativeValue;
};
