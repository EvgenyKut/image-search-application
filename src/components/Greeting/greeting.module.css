.wrapper {
  display: flex;
  font-size: 20px;
  align-items: center;
}

.avatar {
  margin-right: 0.5rem;
}

.lightTheme {
  display: none;
}

.darkTheme {
  display: none;
}

body[data-theme='light'] .lightTheme {
  display: block;
}

body[data-theme='dark'] .darkTheme {
  display: block;
}
