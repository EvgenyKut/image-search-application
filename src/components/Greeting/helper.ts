export const greetingFormatting = (name: string, time: number) => {
  let timePeriod = '';

  if (time >= 22 || time < 6) {
    timePeriod = 'night';
  } else if (time >= 18) {
    timePeriod = 'evening';
  } else if (time >= 12) {
    timePeriod = 'afternoon';
  } else if (time >= 6) {
    timePeriod = 'morning';
  }
  return name ? `Good ${timePeriod}, ${name}!` : `Good ${timePeriod}!`;
};
