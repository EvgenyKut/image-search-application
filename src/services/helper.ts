import { IImage, IImageRaw } from '../models/IImage';

export const normalizeImages = (images: IImageRaw[]): IImage[] => {
  const sortedImages = images.sort((a, b) => b.likes - a.likes);
  const normalizedImages: IImage[] = sortedImages.map((image: IImageRaw) => ({
    id: image?.id || 0,
    url: image?.urls?.small || '',
    alt: image?.alt_description || '',
  }));
  return normalizedImages;
};
