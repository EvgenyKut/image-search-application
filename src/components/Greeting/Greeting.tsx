import React from 'react';
import styles from './greeting.module.css';
import { greetingFormatting } from './helper';
import Avatar from '@mui/material/Avatar';
import { useAppSelector } from '../../hooks';
import { capitalizeFirstLetter } from '../../helpers/helper';

type GreetingProps = {
  name: string;
};
const Greeting: React.FC<GreetingProps> = ({ name }) => {
  const { isAuth } = useAppSelector((state) => state.authReducer);
  return (
    <div className={styles.wrapper} data-testid="greeting">
      {isAuth && (
        <Avatar className={styles.avatar}>
          {capitalizeFirstLetter(name.slice(0, 1))}
        </Avatar>
      )}
      {greetingFormatting(name, new Date().getHours())}
    </div>
  );
};

export default Greeting;
