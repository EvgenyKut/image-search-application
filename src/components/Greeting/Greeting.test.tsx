import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Greeting from './index';
import { greetingFormatting } from './helper';

describe('Greeting-component render test', () => {
  it('should behave correctly', () => {
    const value = greetingFormatting('Evgeny', new Date().getHours());
    render(<Greeting name={'Evgeny'} />);
    const textField = screen.getByTestId('greeting') as HTMLInputElement;
    expect(textField).toBeInTheDocument();
    expect(textField).toBeVisible();
    expect(textField.textContent).toBe(value);
  });
});
