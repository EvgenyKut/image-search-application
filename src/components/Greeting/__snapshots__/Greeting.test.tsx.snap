// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Greeting-component render test should have snapshot 1`] = `
<div
  className="wrapper"
  data-testid="greeting"
>
  Good afternoon, Evgeny!
</div>
`;
